app/controllers/api/predictions_controller.rb

```diff
@@ -26,12 +43,13 @@ def authenticate_by_api_token
       @user = User.find_by_api_token(params[:api_token])
 
       unless valid_params_and_user?
-        render json: invalid_message, status: :unauthorized
+        render json: invalid_api_message, status: :unauthorized
       end
     end
 
-    def valid_params_and_user?
-      params[:api_token] && @user
+    def authorized_to_view_prediction?
+      return true unless @prediction.private?
```

Comment:
This logic doesn't feel quite right. As a user, I would be surprised to find out I can't look at my own private predictions via the API. I think this should be rolled into the authorised_for method so all auth checks go through that.